addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -43,6 +41,43 @@ def _remove_misleading_values(self, **kwargs):
                 if current_value and current_value != value:
                     self[field] = False
 
+    @api.model
+    def _setup_refactored_picking_package_levels(self, picking, move_lines):
+        """
+        Identify any package levels set for the supplied move lines.
+
+        If the package levels exclusively contain the move lines, then simply move all
+        package links to the new refactored picking.
+
+        Otherwise create a new package level for each unique package within the move lines,
+        and set the relevant move lines to point to this package.
+
+        By setting the package levels as needed we avoid extra package levels potentially
+        being created later on via `_check_entire_pack`. This then also avoids the potential for
+        erroneous moves being created when `_generate_moves` is called on a newly generated
+        package level.
+        """
+        PackageLevel = self.env["stock.package_level"]
+
+        move_lines_with_package_levels = move_lines.filtered("package_level_id")
+        package_levels = move_lines_with_package_levels.mapped("package_level_id")
```

Comment:
I don't think we need `mapped` for relational fields anymore (same for line 66).

Reply:
Nope, old habits... fixed.